File: Scripts/results/generated_code_results/baseline_gpt/News&Information/CNN/LogInScreen.js
import React, { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, StyleSheet, Image } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

const LogInScreen = ({ navigation }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [hidePassword, setHidePassword] = useState(true);

  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={() => navigation.goBack()}>
        <Icon name="arrow-back" size={28} />
      </TouchableOpacity>
      <Text style={styles.title}>Sign in to your CNN account</Text>
      <View style={styles.signUpContainer}>
        <Text style={styles.text}>Don’t have an account? </Text>
        <TouchableOpacity onPress={() => navigation.navigate('CreateAccount')}>
          <Text style={styles.linkText}>Sign up</Text>
        </TouchableOpacity>
      </View>
      <TextInput
        style={styles.input}
        placeholder="Email address"
        value={email}
        onChangeText={setEmail}
        keyboardType="email-address"
        autoCapitalize="none"
      />
      <View style={styles.passwordContainer}>
        <TextInput
          style={styles.input}
          placeholder="Password"
          value={password}
          onChangeText={setPassword}
          secureTextEntry={hidePassword}
          autoCapitalize="none"
        />
        <TouchableOpacity onPress={() => setHidePassword(!hidePassword)}>
          <Icon name={hidePassword ? 'visibility-off' : 'visibility'} size={24} />
        </TouchableOpacity>
      </View>
      <TouchableOpacity style={styles.signInButton}>
        <Text style={styles.signInButtonText}>Sign in</Text>
      </TouchableOpacity>
      <TouchableOpacity onPress={() => navigation.navigate('ForgotPassword')}>
        <Text style={styles.linkText}>Forgot your password?</Text>
      </TouchableOpacity>
      <View style={styles.dividerContainer}>
        <View style={styles.divider} />
        <Text style={styles.orText}>or</Text>
        <View style={styles.divider} />
      </View>
      <Text style={styles.policyText}>
        By creating an account or signing in, you agree to the{' '}
        <Text style={styles.linkText}>Terms of Use</Text> and have read our{' '}
        <Text style={styles.linkText}>Privacy Policy</Text>. You understand CNN and its affiliates may
        use your email address to provide updates, ads, and offers. You can opt out via the{' '}
        <Text style={styles.linkText}>Privacy Policy</Text>.
      </Text>
      <TouchableOpacity style={styles.googleButton}>
        <Image source={require('../assets/snack-icon.png')} style={styles.googleIcon} />
        <Text style={styles.googleButtonText}>Continue with Google</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: '#fff',
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginVertical: 20,
  },
  signUpContainer: {
    flexDirection: 'row',
    marginBottom: 20,
  },
  text: {
    fontSize: 14,
  },
  linkText: {
    fontSize: 14,
    color: '#007aff',
  },
  input: {
    borderWidth: 1,
    borderColor: '#ddd',
    padding: 10,
    marginVertical: 10,
    borderRadius: 5,
    flex: 1,
  },
  passwordContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  signInButton: {
    backgroundColor: '#888',
    padding: 15,
    alignItems: 'center',
    borderRadius: 5,
    marginVertical: 20,
  },
  signInButtonText: {
    color: '#fff',
    fontWeight: 'bold',
  },
  dividerContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 20,
  },
  divider: {
    flex: 1,
    height: 1,
    backgroundColor: '#ddd',
  },
  orText: {
    marginHorizontal: 10,
    fontSize: 14,
  },
  policyText: {
    fontSize: 12,
    color: '#666',
    marginVertical: 20,
  },
  googleButton: {
    flexDirection: 'row',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: '#ddd',
    padding: 10,
    borderRadius: 5,
  },
  googleIcon: {
    width: 20,
    height: 20,
    marginRight: 10,
  },
  googleButtonText: {
    fontSize: 16,
    color: '#444',
  },
});

export default LogInScreen;

File: Scripts/results/generated_code_results/Flutter/News&Information/CNN/LogInScreen.dart
import 'package:flutter/material.dart';
class LogInScreen extends StatefulWidget {
@override
_LogInScreenState createState() => _LogInScreenState();
}
class _LogInScreenState extends State<LogInScreen> {
bool _obscureText = true;
final TextEditingController _emailController = TextEditingController();
final TextEditingController _passwordController = TextEditingController();
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pushNamed(context, 'Settings'),
),
title: Row(
mainAxisAlignment: MainAxisAlignment.end,
children: [
Icon(Icons.wifi),
SizedBox(width: 8),
Icon(Icons.battery_full),
],
),
),
body: SingleChildScrollView(
padding: EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Icon(Icons.circle, color: Colors.black, size: 40),
SizedBox(height: 16),
Text(
'Sign in to your CNN account',
style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
),
SizedBox(height: 8),
RichText(
text: TextSpan(
style: TextStyle(color: Colors.black),
children: [
TextSpan(text: 'Don't have an account? '),
TextSpan(
text: 'Sign up',
style: TextStyle(color: Colors.blue),
recognizer: TapGestureRecognizer()
..onTap = () {
Navigator.pushNamed(context, 'CreateAccont');
},
),
],
),
),
SizedBox(height: 24),
TextField(
controller: _emailController,
decoration: InputDecoration(
labelText: 'Email address',
border: OutlineInputBorder(),
),
keyboardType: TextInputType.emailAddress,
),
SizedBox(height: 16),
TextField(
controller: _passwordController,
obscureText: _obscureText,
decoration: InputDecoration(
labelText: 'Password',
border: OutlineInputBorder(),
suffixIcon: IconButton(
icon: Icon(_obscureText ? Icons.visibility_off : Icons.visibility),
onPressed: () {
setState(() {
_obscureText = !_obscureText;
});
},
),
),
),
SizedBox(height: 24),
ElevatedButton(
child: Text('Sign in'),
onPressed: () {
// Implement sign in logic
},
style: ElevatedButton.styleFrom(
minimumSize: Size(double.infinity, 50),
),
),
SizedBox(height: 16),
Center(
child: TextButton(
child: Text('Forgot your password?'),
onPressed: () {
// Navigate to password recovery
},
),
),
SizedBox(height: 16),
Row(
children: [
Expanded(child: Divider()),
Padding(
padding: EdgeInsets.symmetric(horizontal: 8),
child: Text('or'),
),
Expanded(child: Divider()),
],
),
SizedBox(height: 16),
OutlinedButton.icon(
icon: Image.asset('assets/images/template.png', height: 24),
label: Text('Continue with Google'),
onPressed: () {
// Implement Google sign in
},
style: OutlinedButton.styleFrom(
minimumSize: Size(double.infinity, 50),
),
),
SizedBox(height: 24),
Text(
'By creating an account or signing in, you agree to the Terms of Use and have read our Privacy Policy. You understand CNN and its affiliates may use your email address to provide updates, ads, and offers. You can opt out via the Privacy Policy.',
style: TextStyle(fontSize: 12, color: Colors.grey),
),
],
),
),
);
}
@override
void dispose() {
_emailController.dispose();
_passwordController.dispose();
super.dispose();
}
}